libtransact/src/execution/adapter/static_adapter.rs

```diff
@@ -0,0 +1,619 @@
+
+        // Create and execute a simple transaction
+        let txn_pair = make_transaction(&[Command::Set {
+            address: "abc".into(),
+            value: b"abc".to_vec(),
+        }]);
+        let context_id = context_manager.create_context(&[], &state_id);
+
+        let (send, recv) = std::sync::mpsc::channel();
+        static_adapter.execute(
+            txn_pair,
+            context_id.clone(),
+            Box::new(move |res| {
+                send.send(res).expect("Unable to send result");
+            }),
+        );
+        let result = recv.recv().unwrap();
+
+        assert_eq!(
+            ExecutionTaskCompletionNotification::Valid(context_id.clone()),
+            result.unwrap()
+        );
+        assert_eq!(
+            vec![("abc".to_owned(), b"abc".to_vec())],
+            context_manager
+                .get(&context_id, &["abc".to_owned()])
+                .unwrap()
+        );
+
+        assert_eq!(true, Box::new(static_adapter).stop());
+    }
+
+    /// Apply the static adapater with a failing tranaction
+    #[test]
+    fn apply_static_adapter_invalid_txn() {
+        let registry = MockRegistry::default();
+
+        let state = HashMapState::new();
+        let state_id = HashMapState::state_id(&HashMap::new());
+
+        let mut context_manager: ContextManager = ContextManager::new(Box::new(state));
+
+        let handler = CommandTransactionHandler::new();
+
+        let mut static_adapter =
+            StaticExecutionAdapter::new_adapter(vec![Box::new(handler)], context_manager.clone())
+                .expect("Could not create adapter");
+
+        static_adapter.start(Box::new(registry.clone()));
+
+        // Create and execute a failing transaction.
+        let txn_pair = make_transaction(&[
+            Command::Get {
+                address: "abc".into(),
+            },
+            Command::Fail {
+                error_msg: "Test Fail Succeeded".into(),
+            },
+        ]);
+
+        let txn_id = txn_pair.transaction().header_signature().to_owned();
+        let context_id = context_manager.create_context(&[], &state_id);
+
+        let (send, recv) = std::sync::mpsc::channel();
+        static_adapter.execute(
+            txn_pair,
+            context_id.clone(),
+            Box::new(move |res| {
+                send.send(res).expect("Unable to send result");
+            }),
+        );
+        let result = recv.recv().unwrap();
+
+        assert_eq!(
+            ExecutionTaskCompletionNotification::Invalid(
+                context_id,
+                InvalidTransactionResult {
+                    transaction_id: txn_id,
+                    error_message: "Test Fail Succeeded".into(),
+                    error_data: vec![],
+                }
+            ),
+            result.unwrap()
+        );
+
+        assert_eq!(true, Box::new(static_adapter).stop());
+    }
+
+    fn make_transaction(commands: &[Command]) -> TransactionPair {
```

Comment:
Suggest we put the Command handler/txn creation into the workload module, even if it doesn't currently implement the traits there.

Reply:
Addressed.